Modules/mathmodule.c

```diff
@@ -2189,8 +2205,7 @@ loghelper(PyObject* arg, double (*func)(double))
 
         /* Negative or zero inputs give a ValueError. */
         if (!_PyLong_IsPositive((PyLongObject *)arg)) {
-            PyErr_SetString(PyExc_ValueError,
-                            "math domain error");
+            PyErr_Format(PyExc_ValueError, err_msg, arg);
```

Comment:
Danger! The argument can be arbitrary negative integer. Creating it decimal representation has superlinear complexity. Even if it could have linear complexity, it can be too long, and the result can be too long. Oh, and normally you will get a ValueError due to the int_max_str_digits limit. In any case the cost of formatting the error message can be too high.

cc @tim-one

Reply:
Yes, I'll refactor loghelper() case.  Probably we can omit argument value in errors from log().  The only problem is that it (unfortunately) has 2-arg form too.